.scalafmt.conf

```diff
@@ -1,7 +1,7 @@
 runner.dialect = scala212
 
 # Version is required to make sure IntelliJ picks the right version
-version = 3.5.9
+version = 3.8.3
```

Comment:
Any reason updating the version?

Reply:
Thanks @zhztheplayer for reviewing. When using the older version, the CI ran into exception
```
2024-08-08T00:57:49.5831318Z [INFO] --- spotless-maven-plugin:2.27.2:check (spotless-check) @ gluten-ut-spark35 ---
2024-08-08T00:57:50.3039149Z [ERROR] Step 'scalafmt' found problem in 'src/test/scala/org/apache/gluten/utils/clickhouse/ClickHouseTestSettings.scala':
2024-08-08T00:57:50.3040518Z null
2024-08-08T00:57:50.3040881Z java.lang.StackOverflowError
2024-08-08T00:57:50.3101797Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3102978Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3104855Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3108247Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3110226Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3113485Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3116925Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3125982Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3132026Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3133287Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3134730Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3136020Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3137364Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3138459Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3139722Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3141082Z     at scala.meta.transversers.Traverser.applyTerm (Traverser.scala:4)
2024-08-08T00:57:50.3142179Z     at scala.meta.transversers.Traverser.apply (Traverser.scala:4)
2024-08-08T00:57:50.3143475Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
...
...
2024-08-08T00:57:50.3676975Z     at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply (FormatWriter.scala:1568)
2024-08-08T00:57:50.3679088Z Exception in thread "main" java.lang.StackOverflowError
2024-08-08T00:57:50.3680068Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
2024-08-08T00:57:50.3681434Z 	at scala.meta.transversers.Traverser.apply(Traverser.scala:4)
2024-08-08T00:57:50.3682658Z 	at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply(FormatWriter.scala:1568)
2024-08-08T00:57:50.3683915Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
2024-08-08T00:57:50.3684924Z 	at scala.meta.transversers.Traverser.apply(Traverser.scala:4)
2024-08-08T00:57:50.3686232Z 	at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply(FormatWriter.scala:1568)
2024-08-08T00:57:50.3687447Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
2024-08-08T00:57:50.3688408Z 	at scala.meta.transversers.Traverser.apply(Traverser.scala:4)
2024-08-08T00:57:50.3689522Z 	at org.scalafmt.internal.FormatWriter$FormatLocations$$anon$2.apply(FormatWriter.scala:1568)
2024-08-08T00:57:50.3690706Z 	at scala.meta.transversers.Traverser.applyTerm(Traverser.scala:4)
```

This happened consistently and multiple tries of
`mvn spotless:apply -P....` didn't help, we tried both Velox and Clickhouse profiles with UTs.

Upgrading the version helped.

Reply:
Here's the failure CI link - https://github.com/apache/incubator-gluten/actions/runs/10280644913/job/28490950203?pr=6326